src/HealthGPS/risk_factor_adjustable_model.cpp

```diff
@@ -31,11 +32,18 @@ struct FirstMoment {
 namespace hgps {
 
 RiskFactorAdjustableModel::RiskFactorAdjustableModel(
-    const RiskFactorSexAgeTable &risk_factor_expected)
-    : risk_factor_expected_{risk_factor_expected} {}
-
-const RiskFactorSexAgeTable &RiskFactorAdjustableModel::get_risk_factor_expected() const noexcept {
-    return risk_factor_expected_;
+    std::shared_ptr<RiskFactorSexAgeTable> expected,
+    std::shared_ptr<std::unordered_map<core::Identifier, double>> expected_trend)
+    : expected_{std::move(expected)}, expected_trend_{std::move(expected_trend)} {}
+
+double RiskFactorAdjustableModel::get_expected(RuntimeContext &context, core::Gender sex, int age,
+                                               const core::Identifier &factor) const noexcept {
+    double expected = expected_->at(sex, factor).at(age);
+    if (expected_trend_->contains(factor)) {
+        int elapsed_time = context.time_now() - context.start_time();
+        expected *= pow(expected_trend_->at(factor), elapsed_time);
```

Comment:
Are these arguments the wrong way round? I'm trying to wrap my head round why you'd raise to the power of `elapsed_time`.

Reply:
Yeah, it' look a bit strange, but it's basically an exponential trend in time, With time factor `t`, we have `with_trend = without_trend * t^elapsed`. So for year zero, we have `with_trend = without_trend * t^0 = without_trend`, and `t` factor changes exponentially with years.

Reply:
That reminds me, I need to bound these values...